setup.cfg

```diff
@@ -81,14 +81,14 @@ setup_requires =
 #####################################################################################################
 install_requires =
     alembic>=1.5.1, <2.0
-    argcomplete~=1.10
+    argcomplete>=1.10
     attrs>=20.0, <21.0
     blinker
     cached_property~=1.5;python_version<="3.7"
     cattrs~=1.1, !=1.7.*
     # Required by vendored-in connexion
     clickclick>=1.2
-    colorlog>=4.0.2, <6.0
+    colorlog>=4.0.2
```

Comment:
And `<7`? WDYT?

Reply:
No strong opinion, I went back and forth with it before creating the PR but remove those as we recommend using constraints for installation anyway so keep the deps "open". 

Pushed https://github.com/apache/airflow/pull/20878/commits/a83dba737b8685e4e9870af3b0ee1fc5cad82eb3 to add those upper bounds

Reply:
"we recommend using constraints" -- recommend but don't require.

Reply:
@ashb We actually only support installing airlfow with constraints. Any other use is not supported and we very clearly write about it in our https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html

> While there are some successes with using other tools like poetry or pip-tools, they do not share the same workflow as pip - especially when it comes to constraint vs. requirements management. Installing via Poetry or pip-tools is not currently supported. If you wish to install airflow using those tools you should use the constraints and convert them to appropriate format and workflow that your tool requires.

The way how we have "open" upper bound requirements for most of our dependencies pretty much mandates using constraints. And as far as I know there is no other solution if we do not want our users to be limited. 

Unless we are able to define which constraints (and why) shoudl be limited, and which are not, I think deciding on some of those to be upper-limited is really arbitrary.

I think we need to agree on some rules there rather than have different approach for different PRs/Constraints.

I will ask a discussion on the devlist.